src/metabase/native_query_analyzer.clj

```diff
@@ -30,37 +29,49 @@
        (or (not config/is-test?)
            *parse-queries-in-test?*)))
 
-(defn- normalize-name
-  ;; TODO: This is wildly naive and will be revisited once the rest of the plumbing is sorted out
-  ;; c.f. Milestone 3 of the epic: https://github.com/metabase/metabase/issues/36911
-  [name]
-  (-> name
-      (str/replace "\"" "")
-      u/lower-case-en))
-
 (def ^:private field-and-table-fragment
   "HoneySQL fragment to get the Field and Table"
   {:from [[:metabase_field :f]]
    ;; (t2/table-name :model/Table) doesn't work on CI since models/table.clj hasn't been loaded
    :join [[:metabase_table :t] [:= :table_id :t.id]]})
 
+(defn- field-query
+  "Exact match for quoted fields, case-insensitive match for non-quoted fields"
+  [field value]
+  (if (= (first value) \")
+    [:= field (subs value 1 (dec (count value)))]
```

Comment:
it's probably a safe assumption that an opening `"` implies a closing `"` but something about this makes me a little nervous :sweat_smile:

Reply:
are there crazy escape cases to worry about?

Reply:
![image](https://github.com/metabase/metabase/assets/784417/e0e424e5-530a-43bb-8525-775bff37a0cc)
I think the answer is "no"

Reply:
wait yes:

```
french_towns=# create table bar("CaseSensitive" int, CaseInSensitive text, "Perv""e""rse" int);

french_towns=# \d bar
                     Table "public.bar"
     Column      |  Type   | Collation | Nullable | Default 
-----------------+---------+-----------+----------+---------
 CaseSensitive   | integer |           |          | 
 caseinsensitive | text    |           |          | 
 Perv"e"rse      | integer |           |          | 

```

Reply:
Hmmm but if we're getting `"Perv""e""erse"`, then I need to de-escape it, right? I.e. replace double-quotes with single quotes?

Reply:
Oh shoot, good point